sorted_db/db.go

```diff
@@ -136,21 +139,18 @@ func indexByte(s []byte, i, m int, c byte) int {
 	return -1
 }
 
-// Search uses a binary search looking for needle, and returns the full match line.
-// the needle should already have the record separator appended
-func (db *DB) Search(needle []byte) []byte {
-	db.RLock()
-
+// findFirstRecordAfterMatch performs a binary search to find the first record
+// after needle prefix matches. If includeExactMatch is set, returns the first
+// record that exactly matches, if it exists.
```

Comment:
`...and -1 if it does not`

It might be obvious after reading the documentation for `sort.Search`, but might as well be explicit and save the reader some time.